test/signalrclienttests/hub_connection_impl_tests.cpp

```diff
@@ -77,6 +78,32 @@ TEST(start, start_starts_connection)
     ASSERT_EQ(connection_state::connected, hub_connection->get_connection_state());
 }
 
+TEST(start, start_sets_connection_data)
+{
+    web::uri requested_url;
+    auto web_request_factory = std::make_unique<test_web_request_factory>([&requested_url](const web::uri &url)
+    {
+        requested_url = url;
+        return std::unique_ptr<web_request>(new web_request_stub((unsigned short)404, _XPLATSTR("Bad request"), _XPLATSTR("")));
+    });
+
+    hub_connection_impl hub_connection{ _XPLATSTR("http://fakeuri"), _XPLATSTR(""), trace_level::none, std::make_shared<trace_log_writer>(),
+        std::move(web_request_factory), std::make_unique<test_transport_factory>(create_test_websocket_client()) };
+    hub_connection.create_hub_proxy(_XPLATSTR("my_hub"));
+    hub_connection.create_hub_proxy(_XPLATSTR("your_hub"));
+
+    try

```

Comment:
because the connection is not set up to connect successfully. We don't care about that - what we care about is that the url contains connection data.